File: experiments/000-PipelineChecks/cells/python-pytest_1-Empty_free/2024-1c-parcial1/20260921-202040-52205/analysis/own-tests-project/src/simshotels/room.py
class Room:

    RESERVED = "reserved"

    def __init__(self):
        self._guest = None

    # guests

    def guest_type(self):
        return self._guest

    def receive(self, a_guest_type):
        if not self.is_available():
            raise RuntimeError(Room.room_is_not_empty_error_description())
        self._guest = a_guest_type

    def receive_with_reservation(self, a_guest_type):
        if not self.is_reserved():
            raise RuntimeError(Room.room_is_not_reserved_error_description())
        self._guest = a_guest_type

    def reserve(self):
        if not self.is_available():
            raise RuntimeError(Room.room_is_not_empty_error_description())
        self._guest = Room.RESERVED

    # testing

    def is_available(self):
        return self._guest is None

    def is_occupied(self):
        return self._guest is not None

    def is_reserved(self):
        return self._guest == Room.RESERVED

    # accounting

    def profit_using_if_absent_guest_type(self, a_price_list, unknown_guest_type_block):
        if self.is_available():
            return 0
        if self.is_reserved():
            return min(a_price_list.values()) // 2
        if self._guest in a_price_list:
            return a_price_list[self._guest]
        else:
            return unknown_guest_type_block()

    def loss_using_if_absent_guest_type(self, a_price_list, unknown_guest_type_block):
        if self.is_available():
            return max(a_price_list.values())
        return 0

    # error messages (class side)

    @staticmethod
    def room_is_not_empty_error_description():
        return "Room is not empty."

    @staticmethod
    def room_is_not_reserved_error_description():
        return "Room is not reserved."

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2024-1c-parcial1/20260921-205623-93767/workdir/src/main/java/simshotels/Room.java
package simshotels;

import java.util.HashMap;
import java.util.function.Supplier;

public class Room {

    private RoomState state;

    // initialization

    public Room() {
        state = new AvailableRoomState();
    }

    // guests

    public void receive(String aGuestType) {
        state = state.receive(aGuestType);
    }

    public void receiveWithReservation(String aGuestType) {
        state = state.receiveWithReservation(aGuestType);
    }

    public void reserve() {
        state = state.reserve();
    }

    // testing

    public boolean isAvailable() {
        return state.isAvailable();
    }

    public boolean isOccupied() {
        return state.isOccupied();
    }

    public boolean isReserved() {
        return state.isReserved();
    }

    // accounting

    public Integer profitUsingIfAbsentGuestType(HashMap<String, Integer> aPriceList, Supplier<Integer> unknownGuestTypeBlock) {
        return state.profitUsingIfAbsentGuestType(aPriceList, unknownGuestTypeBlock);
    }

    public Integer lossUsingIfAbsentGuestType(HashMap<String, Integer> aPriceList, Supplier<Integer> unknownGuestTypeBlock) {
        return state.lossUsingIfAbsentGuestType(aPriceList, unknownGuestTypeBlock);
    }

    // error messages (class side)

    public static String roomIsNotEmptyErrorDescription() {
        return "Room is not empty.";
    }

    public static String roomIsNotReservedErrorDescription() {
        return "Room is not reserved.";
    }

    // exceptions (class side)

    public static RuntimeException roomIsNotEmptyError() {
        return new RuntimeException(Room.roomIsNotEmptyErrorDescription());
    }

    public static RuntimeException roomIsNotReservedError() {
        return new RuntimeException(Room.roomIsNotReservedErrorDescription());
    }
}
